src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/reverse_ref.cl

```diff
@@ -0,0 +1,117 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "include/batch_headers/data_types.cl"
+#include "include/batch_headers/fetch_data.cl"
+
+#define BATCH_INDEX 0
+#define FEATURE_INDEX 1
+#if INPUT0_DIMS == 4
+    #define Y_INDEX 2
+    #define X_INDEX 3
+#elif INPUT0_DIMS == 5
+    #define Z_INDEX 2
+    #define Y_INDEX 3
+    #define X_INDEX 4
+#elif INPUT0_DIMS == 6
+    #define W_INDEX 2
+    #define Z_INDEX 3
+    #define Y_INDEX 4
+    #define X_INDEX 5
+#endif
+
+
+KERNEL(reverse_ref)(
+        const __global INPUT0_TYPE* input,
+        const __global INPUT1_TYPE* axis,
+        __global OUTPUT_TYPE* output)
+{
+    uint x = get_global_id(0);
+    uint b = (uint)get_global_id(2) % OUTPUT_BATCH_NUM;
+    uint f = (uint)get_global_id(2) / OUTPUT_BATCH_NUM;
```

Comment:
Please use INPUT0_BATCH_NUM because this is for input, even if input and output shapes are same at Reverse OP.

Reply:
done